### PR TITLE
feat: Splicing Bam

### PR DESCRIPTION
**Pull request by ghost**
Added Splicing Bam file as well as updating the test.

### CODE REVIEW

**Comment by hcorrada** [comment]
@VictorAlexRoche please merge main to this branch, that should make it pass test/check

File: src/epivizFileParser/SplicingBamFile.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                            except:
+                                junctions[jxn] = 1
+                except:
+                    pass
+
+        # convert to proper format
+        junction_list = []
+        for key, value in junctions.items():
+            _start, _end = key.split(":")
+            junction_list.append((chr, 0, _end, _start, 0, value))
```
Comment (hcorrada) [inline src/epivizFileParser/SplicingBamFile.py]: The way the junction key is created (L83) suggests it should be this:

```suggestion
            junction_list.append((chr, _start-1, _start, _end, _end+1, value))
```

File: src/epivizFileParser/SplicingBamFile.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        """
+        if self.columns is None:
+            self.columns = ["chr", "region1_start", "region2_start",
+                            "region1_end", "region2_end", "value"]
+        return self.columns
+
+    def _get_junctions(self, chr, start, end):
+        iter = self.file.fetch(chr, start, end)
+
+        wiggle = numpy.zeros(end - start + 1)
```
Comment (hcorrada) [inline src/epivizFileParser/SplicingBamFile.py]: Looks like `wiggle` is not used (not being returned), suggest that everything having to do with this is removed

**Review by hcorrada — CHANGES_REQUESTED** [review]

File: tests/test_SplicingBamFile.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+def test_getRange():
+    (coverage, junctions), error = bb.getRange("chr10", 1, 100000000)
+    assert(error == None)
+
+    assert(len(bb.columns) == 6)
+    assert(bb.columns == ["chr", "region1_start", "region2_start",
+                          "region1_end", "region2_end", "value"])
+    assert(len(coverage) == 8)
+    assert(len(junctions) == 14)
```
Comment (hcorrada) [inline tests/test_SplicingBamFile.py:28]: should test that start/end values equal expected values

**Review by hcorrada — CHANGES_REQUESTED** [review]

File: src/epivizFileParser/SplicingBamFile.py (hunk 1 above)
Comment (ghost) [reply src/epivizFileParser/SplicingBamFile.py]: The junction starts at a point and ends at a point. I don't understand the meaning behind have two points on each start and end. Could there be a range?

On line 83 it is put together to create a unique key, but I am splitting it up for the dataframe returned.

File: src/epivizFileParser/SplicingBamFile.py (hunk 2 above)
Comment (ghost) [reply src/epivizFileParser/SplicingBamFile.py]: I do not know the meaning behind wiggle (a bioinformatics concept?). I was advised to leave it in like it is in the source code.

File: tests/test_SplicingBamFile.py (hunk 3 above)
Comment (ghost) [reply tests/test_SplicingBamFile.py:28]: assertions added for value, start and end

File: src/epivizFileParser/SplicingBamFile.py (hunk 1 above)
Comment (hcorrada) [reply src/epivizFileParser/SplicingBamFile.py]: Let's design for ranges (e.g., that way this representation would cover a splice graph). How it's populated right now (i.e., 1-bp ranges, should not affect the generality of the design. Nonetheless, having 0's there is not a good idea, and the `_start` and `_end` variables look to be out of order.

File: src/epivizFileParser/SplicingBamFile.py (hunk 2 above)
Comment (hcorrada) [reply src/epivizFileParser/SplicingBamFile.py]: You can remove it

**Comment by ghost** [comment]
___________________________________________________________ ERROR collecting tests/test_GtfParsedFile.py ___________________________________________________________
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py:1346: in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/http/client.py:1253: in request
    self._send_request(method, url, body, headers, encode_chunked)
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/http/client.py:1299: in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/http/client.py:1248: in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/http/client.py:1008: in _send_output
    self.send(msg)
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/http/client.py:948: in send
    self.connect()
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/http/client.py:1422: in connect
    self.sock = self._context.wrap_socket(self.sock,
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ssl.py:500: in wrap_socket
    return self.sslsocket_class._create(
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ssl.py:1040: in _create
    self.do_handshake()
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/ssl.py:1309: in do_handshake
    self._sslobj.do_handshake()
E   ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)

During handling of the above exception, another exception occurred:
tests/test_GtfParsedFile.py:12: in <module>
    bb = GtfParsedFile(
.tox/default/lib/python3.9/site-packages/epivizFileParser/GtfParsedFile.py:33: in __init__
    self.file = pd.read_csv(file, sep="\t", names=columns)
.tox/default/lib/python3.9/site-packages/pandas/util/_decorators.py:311: in wrapper
    return func(*args, **kwargs)
.tox/default/lib/python3.9/site-packages/pandas/io/parsers/readers.py:586: in read_csv
    return _read(filepath_or_buffer, kwds)
.tox/default/lib/python3.9/site-packages/pandas/io/parsers/readers.py:482: in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
.tox/default/lib/python3.9/site-packages/pandas/io/parsers/readers.py:811: in __init__
    self._engine = self._make_engine(self.engine)
.tox/default/lib/python3.9/site-packages/pandas/io/parsers/readers.py:1040: in _make_engine
    return mapping[engine](self.f, **self.options)  # type: ignore[call-arg]
.tox/default/lib/python3.9/site-packages/pandas/io/parsers/c_parser_wrapper.py:51: in __init__
    self._open_handles(src, kwds)
.tox/default/lib/python3.9/site-packages/pandas/io/parsers/base_parser.py:222: in _open_handles
    self.handles = get_handle(
.tox/default/lib/python3.9/site-packages/pandas/io/common.py:608: in get_handle
    ioargs = _get_filepath_or_buffer(
.tox/default/lib/python3.9/site-packages/pandas/io/common.py:311: in _get_filepath_or_buffer
    with urlopen(req_info) as req:
.tox/default/lib/python3.9/site-packages/pandas/io/common.py:211: in urlopen
    return urllib.request.urlopen(*args, **kwargs)
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py:214: in urlopen
    return opener.open(url, data, timeout)
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py:517: in open
    response = self._open(req, data)
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py:534: in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py:494: in _call_chain
    result = func(*args)
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py:1389: in https_open
    return self.do_open(http.client.HTTPSConnection, req,
/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py:1349: in do_open
    raise URLError(err)
E   urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)>

**Comment by hcorrada** [comment]
You can skip tests that have remote connections by adding `pytestmark = pytest.mark.remote` at the top of the file. See any of the `test_*_remote` files for an example.

File: tests/test_SplicingBamFile.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            10266332, 10266646, 10266743, 10267044, 10267063, 10267142])
+    assert (coverage['end'].tolist() == [10265473, 10266332,
+            10266646, 10266743, 10267044, 10267063, 10267142, 10268169])
+
+    assert (junctions['value'].tolist() == [
+            1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1])
+    assert (junctions['region2_start'].tolist() == ['10266201', '10266312', '10266657', '10266723', '10266972',
+            '10266978', '10267047', '10266897', '10267023', '10267117', '10267120', '10267123', '10267225', '10267692'])
+    assert (junctions['region1_end'].tolist() == ['10266155', '10266296', '10266641', '10266707', '10266961',
+            '10266972', '10267043', '10266881', '10267007', '10267104', '10267117', '10267121', '10267118', '10267690'])
```
Comment (jkanche) [inline tests/test_SplicingBamFile.py]: these should be integers not strings

Comment (ghost) [reply tests/test_SplicingBamFile.py]: fixed

**Pull request merged**